overlord/snapstate/target.go

```diff
@@ -754,3 +818,471 @@ func validatedComponentInfo(path string, si *snap.Info, csi *snap.ComponentSideI
 	}
 	return componentInfo, nil
 }
+
+// updatePlan contains the data that describes an update, including a list of
+// Target structs that represent the snaps that are to be updated.
+type updatePlan struct {
+	// requested is the list of snaps that were requested to be updated. If
+	// RefreshAll is true, then this list should be empty.
+	requested []string
+	// targets is the list of snaps that are to be updated. Note that this list
+	// does not necessarily match the list of snaps in Requested.
+	targets []target
```

Comment:
are we sure there's not code assuming that requested and targets match? can be do something to avoid that risk going forward?

Reply:
I believe so, yes. These eventually map to the same things as the current `names` and `updates` params for `doUpdate` in master now, which are also two slices that might not be the same length right now.

I could remove the `requested` field and add a field to `target` indicating that the target was created because of an explicit request. Although we'd end up just reconstructing the slice later.